lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_playbook_method.rb

```diff
@@ -23,16 +29,41 @@ def execute
       running_in_state_machine? ? check_task_status(task_id) : wait_for_method(task_id)
     end
 
+    def process_result(task = nil)
+      stack_job(task).refresh_ems
+      log_stdout(task)
```

Comment:
Two things:

1. Would `process_result` ever get called with a `nil` task value?  The `.refresh_ems` call will fail since `stack_job(task)` is likely to return nil (unless `stack_job` was previously called and just returning the cached object.)

2. Is there a reason to pass the `task` object to `log_stdout` stead of the `stack_job` object?  (The `task` object is only used to load the `stack_job` as part of the logging.)

If `task` could be `nil` them maybe something like this:
```ruby
def process_result(task = nil)
  job = stack_job(task)
  if job
    job(task).refresh_ems
    log_stdout(job.reload)
  end

  super
end
```

Reply:
`task` should always be there.